BraintreeCore/src/main/java/com/braintreepayments/api/core/AnalyticsEvent.kt

```diff
@@ -8,5 +8,7 @@ internal data class AnalyticsEvent(
     val isVaultRequest: Boolean = false,
     val startTime: Long? = null,
     val endTime: Long? = null,
-    val endpoint: String? = null
+    val endpoint: String? = null,
+    val experiment: String? = null,
+    val buttonRank: Int? = null
```

Comment:
Curious - was there an issue with using delegation or inheritance to not have the base analytics class have module specific properties? 

also is there a good way we can document these properties even though it's internal to the SDK?

Reply:
> was there an issue with using delegation or inheritance to not have the base analytics class have module specific properties?

No, I didn't want to do that as part of this ticket. After looking at the way it is setup, I think grouping the events into modules within this `AnalyticsEvent` class should be sufficient. We use `putOpt` and only add the relevant parameters to the events, so there isn't extra bloat.
IMO The delegation pattern example you've shared would probably be overly verbose for what we are doing here. If there's a cleaner way and more concise way, I'd prefer that.

Reply:
> also is there a good way we can document these properties even though it's internal to the SDK?

I'd think we'd want to break the properties logically into module based properties. Then it'd probably be easier to document them.
Let me address these issues one by one.

Good callouts!

Reply:
Actually, some of these events are shared by multiple modules. So, there isn't necessarily a proper separation. This is probably the best we'll do on this.

For the documentation though, `buttonRank` could be module specific, but `experiment`, though currently limited to one module, could be anything and could be used by any module. Not sure what the best course of action is. I'll probably just keep it as is.